src/main/java/xyz/wagyourtail/jsmacros/client/movement/MovementQueue.java

```diff
@@ -0,0 +1,86 @@
+package xyz.wagyourtail.jsmacros.client.movement;
+
+import net.minecraft.client.network.ClientPlayerEntity;
+import net.minecraft.util.math.Vec3d;
+import xyz.wagyourtail.jsmacros.client.api.classes.Draw3D;
+import xyz.wagyourtail.jsmacros.client.api.classes.PlayerInput;
+import xyz.wagyourtail.jsmacros.client.api.library.impl.FHud;
+
+import java.util.ArrayList;
+import java.util.List;
+
+import static xyz.wagyourtail.jsmacros.client.JsMacros.LOGGER;
+
+public class MovementQueue {
+    private static final List<PlayerInput> queue = new ArrayList<>();
+    private static final List<Vec3d> predictions = new ArrayList<>();
+    private static ClientPlayerEntity player;
+    private static int queuePos = 0;
+    private static boolean reCalcPredictions;
+
+    public static PlayerInput tick(ClientPlayerEntity newPlayer) {
+        if (queuePos == queue.size()) {
+            return null;
+        }
+
+        player = newPlayer;
+
+        Vec3d diff;
+        if (predictions.size() > 0) {
+            diff = new Vec3d(player.getX() - predictions.get(0).getX(), player.getY() - predictions.get(0).getY(), player.getZ() - predictions.get(0).getZ());
+            if (diff.length() > 0.01D) {
+                LOGGER.info("Pred of by x={}, y={}, z={}", diff.getX(), diff.getY(), diff.getZ());
```

Comment:
this seems a bit console spammy

Reply:
Not if my prediction is good :) , but I think I will change this.